src/modules/notifications/notifications.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Body, Controller, Param, Patch, Req, Request, ValidationPipe } from '@nestjs/common';
+import { NotificationsService } from './notifications.service';
+import { ApiBearerAuth, ApiBody, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { MarkNotificationAsReadDto } from './dtos/mark-notification-as-read.dto';
+
+@ApiBearerAuth()
+@ApiTags('Notification')
+@Controller('notifications')
+export class NotificationsController {
+  constructor(private readonly notificationsService: NotificationsService) {}
+
+  @Patch('/:notificationId/read')
+  @ApiBody({ type: MarkNotificationAsReadDto, description: 'Read status of the notification' })
+  @ApiResponse({ status: 200, description: 'Notification marked as read successfully' })
+  @ApiResponse({ status: 500, description: 'Internal server error' })
+  @ApiResponse({ status: 400, description: 'Bad request' })
+  @ApiResponse({ status: 400, description: 'Notification not found' })
```

Comment:
Add a `MarkNotificationAsReadErrorDto` as a type

e.g

```suggestion
@ApiCreatedResponse({ type: CreateNotificationResponseDto, description: 'Notification created successfully' })
  @ApiUnauthorizedResponse({ type: MarkNotificationAsReadErrorDto, description: 'Unauthorized' })
  @ApiBadRequestResponse({ type: MarkNotificationAsReadErrorDto, description: 'Bad Request' })
  @ApiInternalServerErrorResponse({ type: MarkNotificationAsReadErrorDto, description: 'Internal Server Error' })
```